test/com/google/javascript/jscomp/CollapsePropertiesTest.java

```diff
@@ -76,6 +80,11 @@ public void testIncrement() {
   public void testObjLitDeclaration() {
     test("var a = {b: {}, c: {}}; var d = a.b; var e = a.c",
          "var a$b = {}; var a$c = {}; var d = a$b; var e = a$c");
+
+    test("var a = {b: {}, /** @nocollapse */ c: {}}; "
```

Comment:
Sounds good, let's keep it like this for now.

FYI, in the new type inference we only handle obj lits declared with const as namespaces. (It's both easier to implement, and it's clearer for the programmer to explicitly declare a namespace when they use const.)